pkg/apis/garden/v1beta1/types.go

```diff
@@ -321,6 +321,63 @@ const (
 	MachineImageCoreOS MachineImageName = "CoreOS"
 )
 
+////////////////////////////////////////////////////
+//                    PROJECTS                    //
+////////////////////////////////////////////////////
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Project holds certain properties about a Gardener project.
+type Project struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines the project properties.
+	// +optional
+	Spec ProjectSpec `json:"spec,omitempty"`
+	// Most recently observed status of the Project.
+	// +optional
+	Status ProjectStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ProjectList is a collection of Projects.
+type ProjectList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// Items is the list of Projects.
+	Items []Project `json:"items"`
+}
+
+// ProjectSpec is the specification of a Project.
+type ProjectSpec struct {
+	// Description is a human-readable description of what the project is used for.
+	// +optional
+	Description *string `json:"description,omitempty"`
+	// Owner is a user name, an email address, or any other identifier of a user owning the project.
+	Owner string `json:"owner"`
```

Comment:
This probably should call `Owners` be of type `[]rbacv1.Subject`

https://github.com/kubernetes/api/blob/2315b41a07e851014d77070954fc391a9b2d9aa9/rbac/v1/types.go#L71-L88

as a resource could have multiple owners and the owner is not limited to `User`, but it can also be `ServiceAccount` or `Group`

Reply:
Thanks, agreed that the type should be `rbacv1.Subject`. After talking to @holgerkoser we are of the opinion that there currently is - and in the future should also be - only one owner of a project. If we need to extend that at a point in time we can do that. OK?

Reply:
Exactly the same scenario happened with `CRD`'s `Version` field.
In [1.10](https://github.com/kubernetes/apiextensions-apiserver/blob/f584b16eb23bd2a3fd292a027d698d95db427c5d/pkg/apis/apiextensions/v1beta1/types.go#L23-L41) it had only `Version` field. 
In [1.11](https://github.com/kubernetes/apiextensions-apiserver/blob/aaea8ffc60577d42bca6f1ea93a10ec85ec4f54e/pkg/apis/apiextensions/v1beta1/types.go#L23-L57), because the `CRD`s added support for multiple versions of a custom resource, then `Versions` fields was added, deprecating the old `Version` field and having custom logic for that scenario.

I'd rather be on the safe side and support multiple owners right now and don't have to handle the nasty bits of deprecating a field and migrating to list.

Reply:
OK for me, I don't really care whether it is one or many.
@holgerkoser can you comment, please?

Reply:
Do we want to have exactly one or many responsible people for a project? The combination of [pluralistic ignorance](https://en.wikipedia.org/wiki/Pluralistic_ignorance) and the [diffusion of responsibility](https://en.wikipedia.org/wiki/Diffusion_of_responsibility) will lead to a situation where nobody feels responsible or is able to take any actions in case of problems. Therefore we said, there should be always one person (not serviceaccount) which is responsible (the owner) for a certain project.